bun/src/binding.ts

```diff
@@ -116,10 +113,36 @@ const lib = dlopen(binaryPath, {
 		],
 		returns: "u32",
 	},
-});
+};
+
+let binding: ConvertFns<typeof fns> | null = null;
+
+// Load the compiled Zig shared library
+const libPromise = await openLibrary(path.resolve("../bun"), fns);
+
+/**
+ * Call this api to initialize the binding.
+ */
+export async function initBinding(): Promise<void> {
+	const lib = await libPromise;
+	binding = lib.symbols;
+}
 
-export const binding = lib.symbols;
+/**
+ * Get the binding.
+ */
+export function getBinding(): ConvertFns<typeof fns> {
+	if (binding == null) {
+		throw new Error("Binding not initialized. Call init() first.");
+	}
+	return binding;
+}
 
```

Comment:
I don't think we should use this pattern (reinitializable binding), rather just set as const.
This will really simplify usage everywhere.

eg:
```ts
const lib = await openLibrary(...);

export const binding = lib.symbols;
export const close = lib.close;
```

If some consumer really wants to reinitialize a binding/import, they can do something like in this issue:
- https://github.com/oven-sh/bun/issues/12371

Reply:
why not? top-level-await should work

Reply:
oh thanks @wemeetagain , I missed that point. Added top level await and it's much nicer now for the consumer apis